### PR TITLE
Remove dependency on javaagent-bootstrap from javaagent-extension-api

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**

### CODE REVIEW

File: javaagent-api/src/main/java/io/opentelemetry/javaagent/instrumentation/api/internal/ClassLoaderMatcherCacheHolder.java [hunk 1]
```diff
@@ -3,7 +3,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-package io.opentelemetry.javaagent.bootstrap;
+package io.opentelemetry.javaagent.instrumentation.api.internal;
```
Comment (iNikem) [inline javaagent-api/src/main/java/io/opentelemetry/javaagent/instrumentation/api/internal/ClassLoaderMatcherCacheHolder.java:6]: It still lives in bootstrap classloader, right?

Comment (mateuszrzeszutek) [reply javaagent-api/src/main/java/io/opentelemetry/javaagent/instrumentation/api/internal/ClassLoaderMatcherCacheHolder.java:6]: Yes - but at least it's not in the (internal) `javaagent-bootstrap` module.

Comment (iNikem) [reply javaagent-api/src/main/java/io/opentelemetry/javaagent/instrumentation/api/internal/ClassLoaderMatcherCacheHolder.java:6]: Why exactly do we consider bootstrap module as "internal"?

**Review by iNikem — APPROVED** [review]

File: javaagent-api/src/main/java/io/opentelemetry/javaagent/instrumentation/api/internal/ClassLoaderMatcherCacheHolder.java (hunk 1 above)
Comment (mateuszrzeszutek) [reply javaagent-api/src/main/java/io/opentelemetry/javaagent/instrumentation/api/internal/ClassLoaderMatcherCacheHolder.java:6]: There've been mentions of making it internal: https://github.com/open-telemetry/opentelemetry-java-instrumentation/discussions/2779#discussioncomment-601705

**Review by trask — APPROVED** [review]
👍

**Pull request merged**